shared/util/codeql/util/Location.qll

```diff
@@ -0,0 +1,35 @@
+/** Provides classes for working with locations. */
+
+/**
+ * A location as given by a file, a start line, a start column,
+ * an end line, and an end column.
+ *
+ * For more information about locations see [Locations](https://codeql.github.com/docs/writing-codeql-queries/providing-locations-in-codeql-queries/).
+ */
+signature class LocationSig {
+  /** Gets the 1-based line number (inclusive) where this location starts. */
```

Comment:
I guess you "forgot" `getFile` on purpose, right?

Reply:
Yes, because that would require a `File` class in scope.